homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -353,6 +353,8 @@ async def async_alarm_trigger(self, code: str | None = None) -> None:
         No code needed, a trigger time of zero for the current state
         disables the alarm.
         """
+        if self._state == AlarmControlPanelState.TRIGGERED:
+            return
```

Comment:
1. The dict lookup is by `self._active_state` so wouldn't you want to use that instead of `self._state` to prevent the `KeyError`?

2. I suggest using `SUPPORTED_PRETRIGGER_STATES` here as that signals the intent better.

```suggestion
        if (
            self._active_state not in SUPPORTED_PRETRIGGER_STATES
            or not self._trigger_time_by_state[self._active_state]
        ):
            return
```

Reply:
> Using SUPPORTED_PRETRIGGER_STATES is more self explanatory of why such check is even required, but I prioritized robustness.

How about adding a test which checks that `SUPPORTED_PRETRIGGER_STATES` has all states except `AlarmControlPanelState.TRIGGERED`, with a comment explaining why the test is added and what needs to be checked if it fails?